### PR TITLE
Enable `peer_id` update when establishing the chain

### PR DESCRIPTION
**Pull request by angrybayblade**
## Proposed changes

This PR

- Updates com server
  - `GET /params` endpoint to return `peer_id` for the local tendermint node
  - `POST /params` endpoint to update `persistent_peers` when updating the chain config
- Updates the `RegistrationStartupBehaviour` and `TendermintHandler` to work with the `peer_id` update

## Fixes

If it fixes a bug or resolves a feature request, be sure to link to that issue.

## Types of changes

What types of changes does your code introduce? (A **breaking change** is a fix or feature that would cause existing functionality and APIs to not work as expected.)
_Put an `x` in the box that applies_

- [ ] Non-breaking fix (non-breaking change which fixes an issue)
- [ ] Breaking fix (breaking change which fixes an issue)
- [ ] Non-breaking feature (non-breaking change which adds functionality)
- [ ] Breaking feature (breaking change which adds functionality)
- [ ] Refactor (non-breaking change which changes implementation)
- [ ] Messy (mixture of the above - requires explanation!)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../blob/main/CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `main` branch (left side). Also you should start your branch off our `main`.
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have locally run services that could be impacted and they do not present failures derived from my changes

### CODE REVIEW

File: deployments/Dockerfiles/tendermint/app.py [hunk 1]
```diff
@@ -82,6 +84,43 @@ def override_config_toml() -> None:
...
 
+def update_peers(validators: List[Dict]) -> None:
+    """Fix peers."""
+
+    config_path = Path(os.environ["TMHOME"]) / "config" / "config.toml"
+    config_text = config_path.read_text(encoding="utf-8")
+
+    new_peer_string = 'persistent_peers = "'
+    for peer in validators:
+        new_peer_string += peer["peer_id"] + "@" + peer["hostname"] + ":" + "26656,"
```
Comment (0xArdi) [inline deployments/Dockerfiles/tendermint/app.py:95]: This assumes the p2p port is always 26656

Comment (angrybayblade) [reply deployments/Dockerfiles/tendermint/app.py:95]: Yes, this is the issue I was referring. We can introduce a param arg for this.

File: packages/valory/agents/registration_start_up/tests/test_registration.py [hunk 2]
```diff
@@ -107,6 +107,16 @@ class RegistrationStartUpTestConfig(UseRegistries, UseACNNode, BaseTestEnd2End):
...
 
+    def __set_configs(self, i: int, nb_agents: int) -> None:
+        """Set the current agent's config overrides."""
+        super().__set_configs(i=i, nb_agents=nb_agents)
+
+        self.set_config(
+            dotted_path=f"{self.__args_prefix}.tendermint_p2p_port",
+            value=self._tendermint_image.get_p2p_port(i=i),
+            type_="int",
+        )
```
Comment (angrybayblade) [inline packages/valory/agents/registration_start_up/tests/test_registration.py:118]: The registration startup agent requires this parameter for e2e to pass

File: deployments/Dockerfiles/tendermint/app.py [hunk 3]
```diff
@@ -82,6 +85,48 @@ def override_config_toml() -> None:
         fp.write(config)
 
 
+def update_peers(validators: List[Dict]) -> None:
+    """Fix peers."""
```
Comment (angrybayblade) [inline deployments/Dockerfiles/tendermint/app.py:89]: Method to update persistent peers after genesis file update

File: deployments/Dockerfiles/tendermint/app.py [hunk 4]
```diff
@@ -82,6 +85,48 @@ def override_config_toml() -> None:
...
+
+    config_path = Path(os.environ["TMHOME"]) / "config" / "config.toml"
+    config_text = config_path.read_text(encoding="utf-8")
+
+    new_peer_string = 'persistent_peers = "'
+    for peer in validators:
+        hostname = peer["hostname"]
+        if hostname in ("localhost", "0.0.0.0"):
+            hostname = DOCKER_INTERNAL_HOST
```
Comment (angrybayblade) [inline deployments/Dockerfiles/tendermint/app.py:98]: This node will be running in a docker container and no other node will be running in the same container, if. we receive either `localhost` or `0.0.0.0`, we make an assumption that the address belongs to a node running on the same machine with a different docker container and different p2p port so we replace the hostname with the docker's internal host url

**Review by 0xArdi — DISMISSED** [review]

**Review by 0xArdi — APPROVED** [review]

File: deployments/Dockerfiles/tendermint/app.py (hunk 4 above)
Comment (DavidMinarsch) [reply deployments/Dockerfiles/tendermint/app.py:97]: This should be a comment in code!

File: deployments/Dockerfiles/tendermint/app.py [hunk 5]
```diff
@@ -156,7 +202,13 @@ def get_params() -> Dict:
             )
             priv_key_data = json.loads(priv_key_file.read_text(encoding=ENCODING))
             del priv_key_data["priv_key"]
-            return {"params": priv_key_data, "status": True, "error": None}
+            status = requests.get("http://localhost:26657/status").json()
```
Comment (DavidMinarsch) [inline deployments/Dockerfiles/tendermint/app.py:205]: constant, define above

**Review by DavidMinarsch — APPROVED** [review]

**Pull request merged**